src/App.tsx

```diff
@@ -15,48 +18,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort(
+    (prevGood, nextGood) => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return prevGood.length - nextGood.length;
+        case SortType.ALPHABET:
+          return prevGood.localeCompare(nextGood);
+        default:
+          return 0;
+      }
+    },
+  );
+
+  return !isReversed
+    ? visibleGoods
+    : visibleGoods.reverse();
+}
+
+export const App = () => {
+  const [isRev, setIsRev] = useState(false);
+  const [sortBy, setSortBy] = useState<SortType>(SortType.NONE);
+  const [goodsList, setGoodsList] = useState<string[]>();
+
+  const options = {
+    sortType: sortBy,
+    isReversed: isRev,
+  };
+
+  useEffect(() => {
+    const reorderedGoods = getReorderedGoods(
+      goodsFromServer,
+      options,
+    );
+
+    setGoodsList(reorderedGoods);
+  }, [goodsList]);
+
+  const handleSort = (sortByProp: SortType) => () => setSortBy(sortByProp);
+
+  const handleSortAlphabeticaly = handleSort(SortType.ALPHABET);
+
+  const handleSortByLength = handleSort(SortType.LENGTH);
+
+  const handleReverse = () => setIsRev((prevIsRev) => !prevIsRev);
+
+  const handleReset = () => {
+    setSortBy(SortType.NONE);
+    setIsRev(false);
+  };
+
+  const reordered = isRev || sortBy !== 0;
```

Comment:
Naming conventions are important in programming because they make code more readable and easier to understand. Here are some common naming conventions for boolean variables and functions:

#### Boolean Variables:

Start with "is", "has", "should", or "does" to indicate that the variable represents a true or false condition.
Use descriptive names that accurately convey the meaning of the variable.
Use camelCase for multi-word names (e.g. "isCompleted" instead of "is_completed").

Examples:
```
isLoggedin
hasPermission
shouldShowDialog
doesExist
```

#### Functions:

Start with a verb that describes the action the function performs.
Use descriptive names that accurately convey the purpose of the function.
Use camelCase for multi-word names.

Examples:
```
handleClick
validateForm
renderListItems
deleteAccount
```

These naming conventions help to make code more readable and self-explanatory, which is especially important when working on larger projects with other developers.

Reply:
had done all desired renaming